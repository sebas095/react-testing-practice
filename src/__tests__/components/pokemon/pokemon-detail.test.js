import { render, shallow } from "enzyme";

import Loader from "../../../components/common/loader.jsx";
import PokemonDetail from "../../../components/pokemon/pokemon-detail";
import {
  PokemonName,
  PokemonImage,
} from "../../../components/pokemon/pokemon-styled";

const props = {
  loading: true,
  pokemon: {
    id: 1,
    name: "Bulbasur",
    types: [{ type: { name: "Grass" } }],
  },
  image: "someURL",
};

describe("Pokemon Detail", () => {
  it("Render ok", () => {
    const wrapper = render(<PokemonDetail {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Show the loading bar", () => {
    const wrapper = shallow(<PokemonDetail {...props} />);
    expect(wrapper.find(Loader)).toHaveLength(1);
  });

  it("Show the pokemon info", () => {
    const mockProps = {
      ...props,
      loading: false,
    };

    const wrapper = shallow(<PokemonDetail {...mockProps} />);
    expect(wrapper.find(PokemonImage)).toHaveLength(1);
    expect(wrapper.find(PokemonName)).toHaveLength(1);
  });
});
