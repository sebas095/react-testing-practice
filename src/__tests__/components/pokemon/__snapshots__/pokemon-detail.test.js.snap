// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Detail Render ok 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-gKsewC",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bdfBwQ jOYFKP",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-gsTCUz cUyFby",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
